### PR TITLE
Add new stanza for ARM aarch64/openmpi

### PR DESCRIPTION
**Pull request by yip111**
TYPE：enhancement

KEYWORDS: ARM 

DESCRIPTION OF CHANGES:
Problem:
There are no ARM-only stanzas in arch/configure.defaults. As the suggestion from https://github.com/wrf-model/WRF/pull/1535 , we need a separate PR for ARM specific update.

Solution:
Edit the arch/configure.defaults file and add a stanza for aarch64 and armv7l, using the GNU compiler options. 

The default optimization is -03, up from the usual -02. On two 36-h nested simulations on a raspberry pi, the difference
is a consistent 3% speed increase. This used CONUS, so the -03 seems robust.

LIST OF MODIFIED FILES:
M arch/configure.defaults

RELEASE NOTE: Added a new stanza that is only for ARM processors (right now aarch64 and armv7l) with GNU.

### CODE REVIEW

**Review by martin-g — DISMISSED** [review]

**Comment by davegill** [comment]
@yip111 
Would you please confirm that the current version of WRF, v4.3, does NOT already build support for ARM.

**Comment by davegill** [comment]
@yip111 
We want to know the following.
Does the existing WRF release v4.3 work with ARM and gfortran, WITHOUT this update?

File: arch/configure.defaults [hunk 1]
```diff
@@ -1980,6 +1980,49 @@ LIB_BUNDLED     = \
...
+FCSUFFIX        =
+BYTESWAPIO      =       -fconvert=big-endian -frecord-marker=4
+FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO)
+FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
+MODULE_SRCH_FLAG =
+TRADFLAG        =      -traditional
+CPP             =      /lib/cpp -P
+AR              =      ar
+ARFLAGS         =      ru
+M4              =      m4 -G
```
Comment (davegill) [inline arch/configure.defaults:2021]: @yip111 
The "-G" removes all GNU extensions to m4. Have you seen troubles with the m4 processor that requires this flag?

File: arch/configure.defaults [hunk 2]
```diff
@@ -1980,6 +1980,49 @@ LIB_BUNDLED     = \
...
+CCOMP           =       gcc
+DM_FC           =       mpif90 -f90=$(SFC)
+DM_CC           =       mpicc -cc=$(SCC) -DMPI2_SUPPORT
+FC              =       CONFIGURE_FC
+CC              =       CONFIGURE_CC
+LD              =       $(FC)
+RWORDSIZE       =       CONFIGURE_RWORDSIZE
+PROMOTION       =       #-fdefault-real-8
+ARCH_LOCAL      =       -DNONSTANDARD_SYSTEM_SUBR  -DWRF_USE_CLM
+CFLAGS_LOCAL    =       -w -O3 -c -march=armv8.2-a
```
[the remark this answers is not in the document]
Comment (yip111) [reply arch/configure.defaults:2002]: Adding this flag enables the WRF to have better performance in the ARM architecture.

**Comment by davegill** [comment]
@yip111 
First of all, we very much appreciate your time. When users provide suggested modifications, it is beneficial to the entire WRF community. There are a couple of questions about this pull request (PR):

1. Without this PR, does the WRF model (branch release-v4.3.1) successfully build on your ARM machine?

2. Regarding the `-march=armv8.2-a` option, I see this in the gcc man page, and `native` looks to be a better choice for software compiled and run on the host architecture.
```
       -march=name
           Specify the name of the target architecture and, optionally,
           one or more feature modifiers.  This option has the form
           -march=arch{+[no]feature}*.

           The permissible values for arch are armv8-a, armv8.1-a,
           armv8.2-a, armv8.3-a, armv8.4-a, armv8.5-a or native.

           The value armv8.5-a implies armv8.4-a and enables compiler
           support for the ARMv8.5-A architecture extensions.

           The value armv8.4-a implies armv8.3-a and enables compiler
           support for the ARMv8.4-A architecture extensions.

           The value armv8.3-a implies armv8.2-a and enables compiler
           support for the ARMv8.3-A architecture extensions.

           The value armv8.2-a implies armv8.1-a and enables compiler
           support for the ARMv8.2-A architecture extensions.

           The value armv8.1-a implies armv8-a and enables compiler
           support for the ARMv8.1-A architecture extension.  In
           particular, it enables the +crc, +lse, and +rdma features.

           The value native is available on native AArch64 GNU/Linux and
           causes the compiler to pick the architecture of the host
           system.  This option has no effect if the compiler is unable
           to recognize the architecture of the host system,
```

3. With the `m4 -G` suggestion, are there GNU extensions to m4 that are causing troubles for you?

**Comment by yip111** [comment]
@davegill 
Sorry I'm late. 
1.Without this pr, WRF model (branch release-v4.3.1) can successfully build on ARM machine.
2.Adding these flags can bring more performance gains,and it also can  provide another reference  option for users.

**Comment by davegill** [comment]
Using the optimization level -03 on arm7l

t2_d02

**Review by davegill — APPROVED** [review]

**Comment by davegill** [comment]
@kkeene44 
Kelly,
No one else is going to review a PR about ARM processors. I have tested this PR on my ARM machine. I have cleaned up the configure.defaults file to be consistent with what we have now as default options. I have modified the commit message to be in final form. Would you please, per this request, approve this additional ARM-processor stanza.

**Review by kkeene44 — APPROVED** [review]

**Comment by martin-g** [comment]
Thank you!

**Pull request merged**